tool/src/java/org/sakaiproject/gradebookng/tool/panels/StudentGradeSummaryGradesPanel.java

```diff
@@ -0,0 +1,279 @@
+		super(id, model);
+
+		this.view = view;
+	}
+	
+	@Override
+	public void onInitialize() {
+		super.onInitialize();
+		
+		//unpack model
+		Map<String,Object> modelData = (Map<String,Object>) this.getDefaultModelObject();
+		String userId = (String) modelData.get("userId");
+		String displayName = (String) modelData.get("displayName");
+		
+		//build the grade matrix for the user
+		final List<Assignment> assignments = this.businessService.getGradebookAssignments();
+        
+		//TODO catch if this is null, the get(0) will throw an exception
+		//TODO also catch the GbException
+		this.gradeInfo = this.businessService.buildGradeMatrix(assignments, Collections.singletonList(userId)).get(0);
+		this.categories = this.businessService.getGradebookCategories();
+
+		final List<String> categoryNames = new ArrayList<String>();
+		final Map<String, List<Assignment>> categoriesToAssignments = new HashMap<String, List<Assignment>>();
+
+		Iterator<Assignment> assignmentIterator = assignments.iterator();
+		while (assignmentIterator.hasNext()) {
+			Assignment assignment = assignmentIterator.next();
+			String category = assignment.getCategoryName() == null ? GradebookPage.UNCATEGORIZED : assignment.getCategoryName();
+
+			if (!categoriesToAssignments.containsKey(category)) {
+				categoryNames.add(category);
+				categoriesToAssignments.put(category, new ArrayList<Assignment>());
+			}
+
+			categoriesToAssignments.get(category).add(assignment);
+		}
+
+		Collections.sort(categoryNames);
+
+		final boolean[] categoryScoreHidden = { false };
+
+		add(new ListView<String>("categoriesList", categoryNames) {
+			private static final long serialVersionUID = 1L;
+
+			@Override
+			protected void populateItem(ListItem<String> categoryItem) {
+				final String category = categoryItem.getModelObject();
+
+				List<Assignment> categoryAssignments = categoriesToAssignments.get(category);
+
+				categoryItem.add(new Label("category", category));
+
+				CategoryDefinition categoryDefinition = null;
+				for (CategoryDefinition aCategoryDefinition : categories) {
+					if (aCategoryDefinition.getName().equals(category)) {
+						categoryDefinition = aCategoryDefinition;
+						break;
+					}
+				}
+
+				if (categoryDefinition != null) {
+					boolean allAssignmentsAreReleased = true;
+					for (Assignment assignment : categoryAssignments) {
+						if (!assignment.isReleased()) {
+							allAssignmentsAreReleased = false;
+							break;
+						}
+					}
+
+					if (isInstructorView() || allAssignmentsAreReleased) {
+						Double score = gradeInfo.getCategoryAverages().get(categoryDefinition.getId());
+						String grade = "";
+						if (score != null) {
+							grade = CategoryColumnCellPanel.formatDouble(score);
+						}
+						categoryItem.add(new Label("categoryGrade", grade));
+
+						String weight = "";
+						if (categoryDefinition.getWeight() == null) {
+							weight = CategoryColumnCellPanel.formatDouble(categoryDefinition.getWeight());
```

Comment:
Yeah if its used in a few places, it can go in the business.util package. There are probably a few of these helpers that can move out and be made static.